macros/src/gen.rs

```diff
@@ -194,11 +194,39 @@ impl<'a> ServiceGenerator<'a> {
                 quote! { None }
             };
 
+            let input_schema = match &handler.arg {
+                Some(PatType { ty, .. }) => {
+                    quote! {
+                        Some(::restate_sdk::discovery::InputPayload {
+                            content_type: Some(<#ty as ::restate_sdk::serde::WithContentType>::content_type().to_string()),
+                            json_schema: Some(<#ty as ::restate_sdk::serde::WithSchema>::generate_schema()),
+                            required: Some(true),
+                        })
+                    }
+                }
+                None => quote! {
+                    Some(::restate_sdk::discovery::InputPayload {
+                        content_type: Some(<() as ::restate_sdk::serde::WithContentType>::content_type().to_string()),
+                        json_schema: Some(<() as ::restate_sdk::serde::WithSchema>::generate_schema()),
+                        required: Some(false),
+                    })
+                }
```

Comment:
When no input, this should be the json manifest: https://github.com/restatedev/restate/blob/main/service-protocol/endpoint_manifest_schema.json#L93

Which I think you can achieve simply creating `InputPayload` with everything `None`